src/bp/ui-admin/src/telemetry.ts

```diff
@@ -112,84 +113,64 @@ export function checkInfoReceived() {
   return !_.includes(info, '')
 }
 
-export function getServerFeedback() {
-  const pkgStr = window.localStorage.getItem('packageToSend')
-  let packages: Array<object> = []
-  if (pkgStr !== null) {
-    packages = JSON.parse(pkgStr)
-  }
-  return packages
-}
+export async function sendServerPackage() {
+  try {
+    const packages = (await api.getSecured().get(serverUrl)).data
 
-export function feedback(pkg) {
-  axios
-    .post(serverUrl, pkg, corsConfig)
-    .then(res => {
-      const packages = getServerFeedback()
-      if (packages.indexOf(pkg) !== -1) {
-        packages.splice(packages.indexOf(pkg), 1)
-        window.localStorage.setItem('packageToSend', JSON.stringify(packages))
-      }
-      console.log(res)
-    })
-    .catch(err => {
-      const packages = getServerFeedback()
-      if (packages.indexOf(pkg) === -1) {
-        packages.push(pkg)
-        window.localStorage.setItem('packageToSend', JSON.stringify(packages))
+    console.log('packages got', packages)
+
+    const feedback = { events: [], status: '' }
+
+    if ((packages !== undefined && _.has(packages, 'url'), _.has(packages, 'events'))) {
+      const url = packages.url
+      const events = packages.events
+
+      let i = 0
+      for (const obj of events) {
+        if (typeof obj === 'string') {
+          events[i] = JSON.parse(obj)
+        }
+        i++
       }
-      console.log(err)
-    })
-}
 
-export function sendServerPackage() {
-  if (window.localStorage.getItem('packageToSend') === null) {
-    window.localStorage.setItem('packageToSend', JSON.stringify([]))
-  }
+      events.map(obj => (obj.source = 'client'))
 
-  const packages = getServerFeedback()
-  packages.forEach((value, index) => {
-    feedback(value)
-  })
+      console.log(events)
 
-  axios
-    .get(serverUrl, corsConfig)
-    .then(res => {
-      if (_.has(res, 'data')) {
-        const payload = res.data.payload
-        const url = res.data.url
-        axios
-          .post(url, payload, corsConfig)
-          .then(res => {
-            feedback({ status: 'OK', data: payload })
-          })
-          .catch(err => {
-            feedback({ status: 'INACCESSIBLE', data: payload })
-            console.log(err)
-          })
+      feedback.events = events.map(obj => obj.uuid)
+
+      try {
+        await axios.post(url, events, corsConfig)
+        feedback['status'] = 'ok'
+        console.log('packages sent', events)
+      } catch (err) {
+        feedback['status'] = 'fail'
+        console.log('Could not send the telemetry packages to the storage server', err)
       }
-    })
-    .catch(err => {
-      console.log(err)
-    })
+    }
+
+    await api
+      .getSecured()
+      .post(serverUrl, feedback)
+      .then(res => console.log('feedbacks sent', res, feedback))
+      .catch(err => console.log('Could not send the feedbacks to the botpress server', err))
```

Comment:
this catch seems noe mandatory since you do a catch at a hire level just next line